wallet/internal/db/itest/block_store_test.go

```diff
@@ -0,0 +1,466 @@
+//go:build itest
+
+package itest
+
+import (
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcwallet/wallet/internal/db"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBlockInsertAndGet tests inserting and retrieving blocks.
+func TestBlockInsertAndGet(t *testing.T) {
+	t.Parallel()
+	ctx := t.Context()
+	store, _ := NewTestStore(t, nil)
+
+	testCases := AllBlockTestCases()
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			err := store.CreateBlock(ctx, tc.Block)
+
+			if tc.ExpectedError != nil {
+				require.Error(
+					t, err, "Expected error for: %s",
+					tc.Name,
+				)
+				require.ErrorIs(t, err, tc.ExpectedError)
+			} else {
+				require.NoError(
+					t, err, "Unexpected error for: %s",
+					tc.Name,
+				)
+				retrieved, err := store.GetBlock(
+					ctx, int32(tc.Block.Height),
+				)
+				require.NoError(t, err)
+				require.Equal(t, tc.Block, retrieved)
+			}
+		})
+	}
+}
+
+// TestGetBlock_NotFound tests the case where a block is not found.
+func TestGetBlockNotFound(t *testing.T) {
+	t.Parallel()
+	ctx := t.Context()
+	store, _ := NewTestStore(t, nil)
+
+	// Attempt to retrieve a block that does not exist.
+	_, err := store.GetBlock(ctx, 999999)
+	require.Error(t, err)
+	require.ErrorIs(t, err, db.ErrBlockNotFound)
+}
+
+// TestGetBlock_QueryFailure tests the case where the underlying database query
+// fails.
+func TestGetBlock_QueryFailure(t *testing.T) {
+	t.Parallel()
+	ctx := t.Context()
+	store, dbConn := NewTestStore(t, nil)
+
+	// Close the database connection to force a query error.
+	require.NoError(t, dbConn.Close())
+
+	// Attempt to retrieve a block, which should now fail.
+	_, err := store.GetBlock(ctx, 1)
+	require.Error(t, err)
+	require.ErrorIs(t, err, db.ErrBlockQuery)
+}
+
+// TestBlockGetBlockTime tests retrieving block timestamps.
+func TestBlockGetBlockTime(t *testing.T) {
+	t.Parallel()
+	ctx := t.Context()
+	store, _ := NewTestStore(t, nil)
+
+	testCases := AllValidBlockTestCases()
+
+	// Insert a few blocks.
+	for i := 0; i < 3 && i < len(testCases); i++ {
+		require.NoError(t, store.CreateBlock(ctx, testCases[i].Block))
+	}
```

Comment:
I think we should explicitly require at least three test cases, rather than silently skipping when they are missing.

```suggestion
    require.GreaterOrEqual(t, len(testCases), 3)
    
	// Insert a few blocks.
	for i := 0; i < 3; i++ {
		require.NoError(t, store.CreateBlock(ctx, testCases[i].Block))
	}
```

Reply:
Updated

Reply:
nit: could remove the && condition in the for loop, as now it's redundant

Reply:
Removed